BridgeEmulator/HueEmulator3.py

```diff
@@ -1577,20 +1585,33 @@ def read_http_request_body(self):
         return b"{}" if self.headers['Content-Length'] is None or self.headers[
             'Content-Length'] == '0' else self.rfile.read(int(self.headers['Content-Length']))
 
+    @staticmethod
+    def extract_version(line: str):
+        if match := re.search(r'Firmware( |(&nbsp;))+([0-9]+).*Bridge( |(&nbsp;))+V2', line, re.IGNORECASE):
+            return int(match.group(3))
+
     def do_POST(self):
         self._set_headers()
         logging.info("in post method")
         logging.info(self.path)
-        self.data_string = self.read_http_request_body()
```

Comment:
We're ignoring request body in /updater code, but reading it takes considerable time (the application sends the whole binary file in it). Therefore I'm only reading the request body in all cases except the /updater.